### ISSUE TITLE
reconnection retries

### ISSUE THREAD

**Issue by clmates** [wesnoth/wesnoth#9498]
### Describe the desired feature

Hi

Sometimes i have to play from locations with bad conection and use my phone to share data with my pc. usually in this scenario i have disconnections and this makes wesnoth to exit from the multiplayyer lobby. usually this are due to micro disconnections. I want to request that instead of disconect you, wesnoth pop up a dialoge offering you to rety the connection or cancel and have the real dc.

thanks

**Comment by clmates** [comment]
Hi, 

Yesterday, as an example I was playing a ladder with StonyDrew and I had like 10 dc. and i come back very quick as they are micro dc, not sure if maybe the timeout period can be adjusted in preferences so the client dont disconnect and just wait a bit more time..... 

Thanks

**Comment by clmates** [comment]
Just complementary information

when the disconnect happens, the system says error waithing for semaphore

**Comment by hengrui** [comment]
Yes I get some kind of weird network issue when playing on windows. Any tip how to debug? I just turned on "Network Log" from the game. but seems like when playing the game it sometimes cause my whole network to become disabled. not sure why! Looking at windows network NCIS it just says disconnected. But only getting this issue with wesnoth and not other games...
